lib/bambora/bank/batch_report_resource.rb

```diff
@@ -79,11 +79,14 @@ def add_messages_to_response(response)
       end
 
       def batch_report_body(request_data)
-        DEFAULT_REQUEST_PARAMS.merge(request_data).merge(
+        additional_data = {
           merchant_id: client.merchant_id,
           pass_code: api_key,
-          sub_merchant_id: client.sub_merchant_id,
-        )
+        }
+
+        params = DEFAULT_REQUEST_PARAMS.merge(additional_data).merge(request_data)
+
+        client.sub_merchant_id.nil? ? params : params.merge(sub_merchant_id: client.sub_merchant_id)
```

Comment:
The suggestion from @cristianoventura is really good here. It may remove the need for the conditional logic, as long as we change the merge order, and then call `#compact`.

```ruby
additional_data = {
  merchant_id: client.merchant_id,
  pass_code: api_key,
  sub_merchant_id: client.sub_merchant_id,
}

DEFAULT_REQUEST_PARAMS.merge(additional_data).merge(request_data).compact
```